src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala

```diff
@@ -330,28 +327,45 @@ class GTGRunner(val masterURL: String,
       nodeMinArea)
 
 
+    edgeListRDD.cache()
+    edgeListRDD.localCheckpoint()
+
     /**
      * Collect the edgeList and construct NodeMap
      */
     val MCCEdgeList = edgeListRDD.collect()
     val MCCNodeMap = createNodeMapFromEdgeList(MCCEdgeList, lat, lon)
 
+    val broadcastedNodeMap = sc.sparkContext.broadcast(MCCNodeMap)
+
     /**
      * Process the edge list. Collect and output edges and vertices
      */
     processEdges(MCCEdgeList, MCCNodeMap)
 
-    val edgeListRDDIndexed = MCCOps.createPartitionIndex(edgeListRDD.collect())
+    /**
+     * Generate the netcdfs
+     */
+    val edgeTupleRDD = edgeListRDD.map({ x =>
+      MCSUtils.get_node_data(x, broadcastedNodeMap.value, lat, lon, false)})
+      .collect()
```

Comment:
@sujen1412 @kwhitehall 
1. Collecting is one way of getting the transformation to actually be evaluated on an RDD. However, we end up also incurring the cost of a "collect" even though nothing is really being communicated back. How about trying to use foreach()?

There's another way to have side effects on RDD's without collecting, and that's through the foreach() function.
Take a look here.
https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/rdd/RDD.scala#L889-L895

I don't see edgeTupleRDD being used anywhere else, so the line of code could just be this :
edgeListRDD.foreach(x => MCSUtils.get_node_data(x, broadcastedNodeMap.value, lat, lon, false))